### PR TITLE
tests(contextmenu): Closes #2445 Test the context menu UI

### LINKED ISSUE #2445: Pinned tiles from Top Sites 2nd row are not visible if Top Sites tiles are displayed on only one row
**[Affected versions]:**
- Firefox 55.0a1 & 52.0.2
- Activity Stream Build: 1.10.0-dev-1492554719-a2769d0

**[Affected Platforms]:**
- All Windows
- All Mac
- All Linux

**[Prerequisites]:**
- Latest Activity Stream dev add-on version was previously installed on a profile with history.
- The "extensions.@activity-streams.experiments.newTabPrefs" pref from about:config is set to true.

**[Steps to reproduce]:**
1. Start the browser with the profile from prerequisites.
2. Open a new tab and click on "Edit" button for Top Sites.
3. Click the "Show more" button.
4. Hover one of the Top Sites tile from the 2nd row and pin it.
5. Click the "Show less" button.
6. Click "Done" button.
 
**[Expected results]:**
- The pinned tab has priority over the others and it is displayed.

**[Actual results]:**
- The pinned tab is hidden and the user will not see that tab unless the Top Sites tiles are displayed on two rows.

**[Notes]:**
- When pinning tabs from the browser, the pinned tabs are displayed in front of the normal ones and are in chronological order.
- Attached a screen recording of the issue:
pin

### PR DESCRIPTION
**Pull request by sarracini**
Adds tests for LinkMenu and ContextMenu UI. Part 2 of #2533

### CODE REVIEW

File: system-addon/test/unit/content-src/components/ContextMenu.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const props = Object.assign({}, DEFAULT_PROPS, {options: [{label: "item1", icon: "icon1"}, {type: "separator"}]});
+    const wrapper = shallow(<ContextMenu {...props} />);
+    assert.equal(wrapper.find(".context-menu-item").props().role, "menuitem");
+  });
+  it("should call onUpdate with false when an option is clicked", done => {
+    const props = Object.assign({}, DEFAULT_PROPS, {
+      options: [{label: "item1"}],
+      onUpdate: val => {
+        assert.isFalse(val);
+        done();
```
Comment (k88hudson) [inline system-addon/test/unit/content-src/components/ContextMenu.test.js]: Can you use `sinon` for this instead?

File: system-addon/test/unit/content-src/components/ContextMenu.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  tabbableOptionsLength: 0
+};
+
+describe("<ContextMenu>", () => {
+  it("shoud be hidden by default", () => {
+    const wrapper = shallow(<ContextMenu {...DEFAULT_PROPS} />);
+    assert.isTrue(wrapper.find(".context-menu").props().hidden);
+  });
+  it("should be visible if props.visible is true", () => {
+    const props = Object.assign({}, DEFAULT_PROPS, {visible: true});
```
Comment (k88hudson) [inline system-addon/test/unit/content-src/components/ContextMenu.test.js]: Up to you whether you want to change it for this file but for the future, I find this is a little more succinct:
```js
const wrapper = shallow(<ContextMenu {...DEFAULT_PROPS} visible={true} />);
```

File: system-addon/test/unit/content-src/components/ContextMenu.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    const wrapper = shallow(<ContextMenu {...props} />);
+    assert.isFalse(wrapper.find(".context-menu").props().hidden);
+  });
+  it("should render all the options provided", () => {
+    const props = Object.assign({}, DEFAULT_PROPS, {options: [{label: "item1"}, {type: "separator"}, {label: "item2"}]});
```
Comment (k88hudson) [inline system-addon/test/unit/content-src/components/ContextMenu.test.js]: Again, I would probably do:
```js
const options = [{label: "item1"}, {type: "separator"}, {label: "item2"}];
const wrapper = shallow(<ContextMenu {...DEFAULT_PROPS} options={options} />);
```

**Comment by as-pine-proxy** [comment]
[Treeherder: c40d02235075ccb1751c91656c0c178347dde1cc](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=c40d02235075ccb1751c91656c0c178347dde1cc)

**Pull request merged**